Assignment6.Rmd

```diff
@@ -70,14 +72,47 @@ table(M2$certified, M2$predict1)
 
 table(M2$certified, M2$predict2)
 
+sum(diag(table(M2$certified, M2$predict1)))/sum(table(M2$certified, M2$predict1))
+
+sum(diag(table(M2$certified, M2$predict2)))/sum(table(M2$certified, M2$predict2))
+
+# The accuracy for the first model is 21.86% and the accuracy for second model is 53.63%, thus model 2 has a lower error rate.
+
 ```
 
 ##Part III
 
 Choose a data file from the (University of Michigan Open Data Set)[https://github.com/bkoester/PLA/tree/master/data]. Choose an outcome variable that you would like to predict. Build two models that predict that outcome from the other variables. The first model should use raw variables, the second should feature select or feature extract variables from the data. Which model is better according to the cross validation metrics?
 
 ```{r}
-
+data2 <- read.csv("student.record.csv", header = T)
+# for the first model, I use SAT scores, ACT scores and HSGPA to predict sex
+data2 <- data2[,c(4:13)]
+data2<- na.omit(data2)
+c.tree3 <- rpart(as.factor(SEX) ~., method = "class", data = data2)
+printcp(c.tree3)
+# for the second model, I use composite SAT scores, composite ACT scores and HSGPA to predict sex
+
+HSGPA <- data2[,1]
+SEX <- data2[, 10]
+ACT <- data2[, 2:6]
+ACT$ACTscore <- rowSums(ACT)
+SAT <- data2[, 7:9]
+SAT$SATscore <- rowSums(SAT)
+data3 <- cbind(HSGPA, ACT, SAT, SEX)
+c.tree4 <- rpart(as.factor(SEX) ~ ACTscore + SATscore + HSGPA, method = "class", data = data3)
+printcp(c.tree4)
+data3$predict1 <- predict(c.tree3, data3, type = "class")
+table(data3$SEX, data3$predict1)
+sum(diag(table(data3$SEX, data3$predict1)))/sum(table(data3$SEX, data3$predict1))
+# the accuracy for the first model is 63.65%.
+
+data3$predict2 <- predict(c.tree4, data3, type = "class")
+table(data3$SEX, data3$predict2)
+sum(diag(table(data3$SEX, data3$predict2)))/sum(table(data3$SEX, data3$predict2))
+# the accuracy for the second model is 58.41%
+
+# Comparing the accuracy, we see that the first model is apparently better than the second one, but considering the parsimony of the model, second model may be better because it used much fewer variables to predict the outcomes.
```

Comment:
Great job! Very clear model comparison. It is very nice that you not only pay attention to the error rates but also model parsimony when selecting model. It would be even better if you would explain the reason why you choose the predictors/features in the second model and give some interpretations about your result.